ibis/expr/format.py

```diff
@@ -246,14 +250,19 @@ def visit(what, extra_indents=0):
         opline = '{}[{}]'.format(opname, type_display)
         return '\n'.join([opline] + formatted_args)
 
-    def _format_subexpr(self, expr):
+    def _format_subexpr(self, expr, get_text_repr=False):
         subexprs = self.memo.subexprs
-        key = expr.op()
+        if get_text_repr:
```

Comment:
Since in this line of code you have the `memo` provided to `Expr._repr(memo=memo)` as `self.memo`, I'm wondering if this would be a bit cleaner if you set an attribute in `memo`, so you don't need to pass `get_text_repr` in so many functions (which makes the code a quite noise IMO).

What do you think?

Reply:
Just changed it, agree it's much cleaner